app/cdap/components/CreateRecipe/index.tsx

```diff
@@ -0,0 +1,150 @@
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { FormEvent, ChangeEvent, useEffect, useState, useRef } from 'react';
+import T from 'i18n-react';
+import { getRecipeByName, createRecipe } from 'components/CreateRecipe/services';
+import { debounce } from 'lodash';
+import { IRecipeFormData } from 'components/RecipeManagement/types';
+import RecipeForm, { CREATE_RECIPE } from 'components/RecipeManagement/RecipeForm';
+import { ICreateRecipeProps } from 'components/RecipeManagement/types';
+
+export default function CreateRecipe({ setShowRecipeForm, setSnackbar }: ICreateRecipeProps) {
+  const [recipeFormData, setRecipeFormData] = useState<IRecipeFormData>({
+    recipeName: '',
+    description: '',
+    directives: [],
+  });
+
+  const [recipeNameErrorMessage, setRecipeNameErrorMessage] = useState<string>('');
+  const [isRecipeNameError, setIsRecipeNameError] = useState<boolean>(false);
+  const [isSaveDisabled, setIsSaveDisabled] = useState<boolean>(true);
+
+  // This static data has to be removed when we have actual API data, then directly we will get that data from store as directives
+  const recipeSteps = ['uppercase: body1', 'titlecase: body2'];
+
+  useEffect(() => {
+    if (
+      recipeFormData.recipeName === '' ||
+      recipeFormData.description === '' ||
+      recipeFormData.recipeName?.trim().length === 0 ||
+      recipeFormData.description?.trim().length === 0 ||
+      isRecipeNameError
+    ) {
+      setIsSaveDisabled(true);
+    } else {
+      setIsSaveDisabled(false);
+    }
+  }, [recipeFormData, isRecipeNameError]);
+
+  const onFormSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    onRecipeDataSave(recipeFormData);
+  };
+
+  const onRecipeDataSave = (recipeFormData: IRecipeFormData) => {
+    const requestBody = {
+      recipeName: recipeFormData.recipeName,
+      description: recipeFormData.description,
+      directives: recipeSteps,
+    };
+    createRecipe(requestBody, onCreateRecipeResponse, onCreateRecipeError);
+  };
+
+  const onCreateRecipeResponse = () => {
+    setIsRecipeNameError(false);
+    setShowRecipeForm(false);
+    setSnackbar({
+      open: true,
+      isSuccess: true,
+      message: `${recipeSteps.length} ${T.translate(
+        'features.WranglerNewUI.RecipeForm.labels.recipeSaveSuccessMessage'
+      )}`,
+    });
+  };
+
+  const onCreateRecipeError = (err) => {
+    if (err.response.message) {
+      setIsRecipeNameError(true);
+    } else {
```

Comment:
Is there a case where `err.response.message` would be empty?

Reply:
removed that check by showing the snackbar with message which we get from response